src/main/java/me/profelements/dynatech/items/electric/MaterialHive.java

```diff
@@ -40,63 +42,52 @@
-                    }
-
-                    if (bee1 != null && bee1.getAmount() == 64 && bee2 != null && bee2.getAmount() == 64) {
-                        SlimefunItem sfBee1 = SlimefunItem.getByItem(bee1);
-                        SlimefunItem sfBee2 = SlimefunItem.getByItem(bee2);
-
-                        if ( sfBee1 != null && sfBee2 != null && sfBee1.getId() == sfBee2.getId()) {
-                            switch (sfBee1.getId()) {
-                                case "BEE":
-                                    return new MachineRecipe(1500, new ItemStack[] {DynaTechItems.BEE, keyInput}, new ItemStack[] {output});
-                                case "ROBOTIC_BEE":
-                                    return new MachineRecipe(900, new ItemStack[] {DynaTechItems.BEE, keyInput}, new ItemStack[] {output});
-                                
-                                case "ADVANCED_ROBOTIC_BEE":
-                                    return new MachineRecipe(600, new ItemStack[] {DynaTechItems.BEE, keyInput}, new ItemStack[] {output});
-                                
-                                default:
-                                    break;
-                            }
-                        } else {
-                            return new MachineRecipe(1800 - secondRemovalAmount, new ItemStack[] {DynaTechItems.BEE, keyInput}, new ItemStack[] {output});
-                        }
-                    } else {
-                        return new MachineRecipe(1800 - secondRemovalAmount, new ItemStack[] {DynaTechItems.BEE, keyInput}, new ItemStack[] {output});
-                    }
-                }
+        if (key == null || key.getAmount() != 64) {
+            return null;
+        }
+        ItemStack output = null;
+        
+        if (vanillaItemsAccepted.getValue().contains(key.getType().toString())) {
+            output = new ItemStack(key.getType());
+        } else {
+            SlimefunItem sfItem = SlimefunItem.getByItem(key);
+            
+            if (sfItem != null && slimefunItemsAccepted.getValue().contains(sfItem.getId())) {
+                output = sfItem.getItem().clone();
+            }
+        }
+        
+        if (output == null) {
+            return null;
+        }
+        
+        int seconds = 1800;
+
+        ItemStack b1 = inv.getItemInSlot(getInputSlots()[0]);
+        
+        if (b1 != null) {
+            SlimefunItem bee1 = SlimefunItem.getByItem(b1);
+            if (bee1 instanceof Bee) {
+                seconds -= ((Bee) bee1).getSpeedMultipler() * b1.getAmount();
+            }
+        }
+        
+        ItemStack b2 = inv.getItemInSlot(getInputSlots()[1]);
+        
+        if (b2 != null) {
+            SlimefunItem bee2 = SlimefunItem.getByItem(b2);
+            if (bee2 instanceof Bee) {
+                seconds -= ((Bee) bee2).getSpeedMultipler() * b2.getAmount();
```

Comment:
This doesn't factor in the change of time when 2 stacks of same bee type are in the MaterialHive.

Reply:
ah ill fix that. How much of a bonus is it supposed to be?

Reply:
Regular bees are 1500, Robotic Bees are 900, Advanced Robotic Bees are 600

Reply:
so if you have 2 regular, it should be 1500 - the speed boosts * stack amounts correct?

Reply:
o i see its only when both are at max stack size aswell

Reply:
yep

Reply:
> so if you have 2 regular, it should be 1500 - the speed boosts * stack amounts correct?

so if you have 2 regular its just 1800 - speed amounts

if you have 2 stacks of regular it just makes the time = 1500

Reply:
i changed it to just subtracting and extra 32 bees worth, because what you had actually made it worse when you had 2 stacks of robotic

Reply:
How does all the bees differ then?

Reply:
I changed some of the base values of each bee, so with 2 stacks of same type its:
normal 1480
robotic 840
advanced robotic 520